src/main/webapp/js/finscholar/appstate/appstate.js

```diff
@@ -51,12 +55,39 @@ class AppState {
     return new AppState();
   }
 
+  /**
+   * Handles updates from the list views.
+   * @param {!Element} node
+   * @private
+   */
+  async listViewUpdate_(node) {
+    if (this.currentView_ instanceof CommonListView) {
+      this.currentView_.removeScrollHandler();
+    }
+    const id = node.id;
+    if (node.classList.contains('colleges')) {
+      this.currentView_ = new CollegePageView();
+    } else {
+      this.currentView_ = new ScholarshipPageView();
+    }
+    this.currentView_.setId(id);
+    await this.currentView_.renderView();
+    this.navbarInstance_ = new NavBar();
+    this.navbarInstance_.registerListener(this.navbarUpdate.bind(this));
```

Comment:
This is identical to below. Abstract away into its own method.

Reply:
done